04-onion-routing.md

```diff
@@ -1177,44 +1179,50 @@ The test vectors use the following parameters:
 	pubkey[3] = 0x032c0b7cf95324a07d05398b240174dc0c2be444d96b159aa6c7f7b1e668680991
 	pubkey[4] = 0x02edabbd16b41c8371b92ef2f04c1185b4f03b6dcd52ba9b78d9d7c89c8f221145
 
-	nhops = 5/20
+	nhops = 5
 	sessionkey = 0x4141414141414141414141414141414141414141414141414141414141414141
-	associated data = 0x4242424242424242424242424242424242424242424242424242424242424242
+
+	failure_source  = node 4
+	failure_message = `incorrect_or_unknown_payment_details`
+      htlc_msat = 100
+      height    = 800000
+      tlv data
+        type  = 34000
```

Comment:
Can you change that to `34001`? We should use an unknown even tlv field, this should be immediately rejected when decoding the message...

Reply:
Why is bad to use an even field? The receiver may understand this field, or pass it up to a higher-level application that interprets the failure message and knows what 34000 is?

Reply:
The spec's tlv requirements say (https://github.com/lightning/bolts/blob/master/01-messaging.md#type-length-value-format):

```
The receiving node:

- otherwise, if type is unknown:
  - if type is even:
    - MUST fail to parse the tlv_stream.
  - otherwise, if type is odd:
    - MUST discard the next length bytes.
```

Eclair does that, which means that to parse this test vector I would need to define that tlv 34000 is a known tlv, which is quite awkward?

Reply:
As mentioned above, it may that only a higher level application knows what 34000 means. So not even decoding the tlv stream when it is received may be too aggressive perhaps.

But updated the test vector nonetheless to 34001.